pkg/controllers/dynakube/extension/tls/reconciler.go

```diff
@@ -36,33 +36,52 @@ func NewReconciler(clt client.Client, apiReader client.Reader, dk *dynakube.Dyna
 }
 
 func (r *reconciler) Reconcile(ctx context.Context) error {
-	query := k8ssecret.Query(r.client, r.client, log)
-
-	if !r.dk.ExtensionsNeedsSelfSignedTLS() {
-		return query.Delete(ctx, &corev1.Secret{ObjectMeta: v1.ObjectMeta{Name: getSelfSignedTLSSecretName(r.dk.Name), Namespace: r.dk.Namespace}})
+	if r.dk.ExtensionsNeedsSelfSignedTLS() {
+		err := r.reconcileSelfSignedMode(ctx)
+		if err != nil {
+			return err
+		}
+	} else {
+		err := r.reconcileTLSRefNameMode(ctx)
+		if err != nil {
+			return err
+		}
 	}
 
+	return nil
+}
+
+func (r *reconciler) reconcileSelfSignedMode(ctx context.Context) error {
```

Comment:
```suggestion
func (r *reconciler) reconcileSelfSigned(ctx context.Context) error {
```
`mode` is a littile bit redundunt here

Reply:
[Fixed](https://github.com/Dynatrace/dynatrace-operator/pull/3843/commits/0d76c6a0da7ba9a83013dbdf06cf46ca01b9ece1)